### ISSUE TITLE
Proposal: Introduce `filter` Attribute for `select`

### ISSUE THREAD

**Issue by sudheer-gowrigari** [openui/open-ui#949]
The proposed `filter` attribute for `select` aims to provide a more efficient, user-friendly way to navigate through long lists of options. This attribute would enable real-time filtering of options based on user input, making it especially useful in scenarios with large datasets.

## Proposed Behavior
The 'filter' attribute would allow the `select` to dynamically filter and display options that match the user's input. This would reduce the number of visible options, helping users to make selections more quickly and accurately.

This proposal also aids the development of the [`filter` attribute for the combobox](https://open-ui.org/components/combobox.explainer/#introducing-filter-attribute)

**Comment by lukewarlow** [comment]
Where would the user input be shown if anywhere?

**Comment by YummyBacon5** [comment]
If this is added, then won't `<selectlist>` become a duplicate of `<combobox>`?

**Comment by sudheer-gowrigari** [comment]
I meant to say: capture user input directly through typing when the `selectlist` is focused.

**Comment by lukewarlow** [comment]
> If this is added, then won't <selectlist> become a duplicate of <combobox>?

For me selectlist is a filterable list for you to pick from. Combobox is a free input with suggestions.

Idk if that's a distinction in the proposal but that's what makes it two different elements in my head.

**Comment by TrevorRice** [comment]
Would there be any affordance given to users to differentiate between the behavior of a filterable and non-filterable `<selectlist>`?

**Comment by lukewarlow** [comment]
@josepharhar can this be closed? Given recent discussions?

**Comment by josepharhar** [comment]
Yeah I think this can be closed in favor of https://github.com/openui/open-ui/issues/931 and https://github.com/openui/open-ui/issues/847

I still don't think we necessarily need two attributes to specify how filtering works
